### PR TITLE
Triples the pressure scale of tank fragmentation. Reworks assembly bombs.

### PR DESCRIPTION
**Pull request by Pickle-Coding**
## About The Pull Request
Scales tank fragmentation pressure scale to 336 atmospheres, from 84 atmospheres. This raises the requirement for the maxcap (5, 10, 20) for a standard TTV to 440 atmospheres, from 160 atmospheres. Removes the additional gas reaction tick before explosion. It also no longer does the pressure tolerance check upon gas merging, so a gas reaction doesn't get skipped. I think that was used for the anomaly refinery, so this breaks it, but it kinda already didn't function like a proper TTV anyways because it skipped a tick. I'll fix this later before I undraft it.

Assembly bombs will perform a heat transfer to the gas mixture with a spark that has the temperature of the igniter's ignition temperature (1,000 Kelvin, 200 Kelvin for condenser), and heat capacity of 300 joules per Kelvin. The tank then processes as normal, which may yield an explosion. Removes the forced explosions based on if checks and shit.

With the new pressure scale value and extra reaction tick removal, this is the reference point of some explosions I tested:
- Plasma and oxygen TTV with T2 thermomachines: 1, 3, 6.
- Plasma and oxygen TTV with 2.7 Kelvin oxygen and 90,000 Kelvin plasma: 2, 4, 9.
- Plasma and tritium + oxygen TTV, plasma at 90,000 Kelvin and tritium + oxygen at 43.15 Kelvin: 9, 19, 39.
- Antinoblium + hypernoblium TTV, antinoblium at 800,000 Kelvin and hypernoblium at 2.7 Kelvin: 35, 71, 143.
- Tritium + Nitrogen TTV, 2.7 Kelvin: 50, 101, 202.
- Tritium + oxygen assembly bomb, 254 Kelvin (lowest temperature that allows it to reach ignition, making it optimal): 3, 7, 15.
- Antinoblium assembly bomb, 2.7 Kelvin: 2, 5, 11.

Various ordnance explosion experiment requirements have been scaled down to reflect these changes.

I know my GBP sucks right now, but I'm opening it early to allow discussion to happen. Drafted because of the potential anomaly refinery issue, and because I want to make it more obvious of how it functions in game so people don't need to code dive or whatever.
## Why It's Good For The Game
The pressure scaling was relatively tiny, only 4 times higher than the minimum needed to consider an explosion, so gas tank explosions were basically either a fail, or easily surpassed the maxcap. It's basically impossible to balance around such a tiny range of pressures.

Assembly bombs based their explosion radii by a bunch of arbitrary if checks, and some hidden power variable. It sucked, because the knowledge was gatekept by codediving or having someone else tell you without explanation as to why. No one could reasonable find out how the assembly bomb works using in game knowledge without needing external sources.

By making the assembly bomb simply heat the contents of the tank, the explosion simply comes from tank fragemntation, which is a consequence of the pressure being really high, which may be a consequence from energetic gas reactions, or letting low specific heat gases get a significant temperature spike. Knowledge of how to make the new assembly bomb mostly comes from your knowledge of atmospherics, rather than knowledge of this specific item.

Ordnance explosion experiment requirements were changed simply so they're not too difficult or impossible due to the tank fragmentation changes.
## Changelog
:cl:
balance: Tank fragmentation pressure scale raised to 336 atmospheres, from 84 atmospheres.
balance: Tank fragmentation no longer causes an extra gas reaction tick before exploding.
balance: Ordnance explosion experiment requirements have been reduced to reflect these changes.
balance: Assembly bombs function based on heating the gases inside the gas tank. They no longer explode based on plasma/oxygen mole and temperature conditions.
/:cl:

### CODE REVIEW

**Comment by moocowswag** [comment]
anomaly refinement requirements

**Comment by necromanceranne** [comment]
If a t2 thermo bomb can still hit the 4 radius minimum, then I agree, its probably still possible to do the bare minimum bomb needed for refining. Certainly, even quick ordnance setups can be a bit of a hurdle for some.

**Comment by Deadgebert** [comment]
What are the mole counts for your tests? Is it just max pressure at your target temperature?
Those mole values and temperatures are probably too hard for toxins to achieve, making this a pretty large nerf to ordnance.

https://docs.google.com/spreadsheets/d/1fXLdJpiBft9i4fg9l3sltsWycbECOWRCj5Xijf406HE/edit#gid=1094049601 you might be able to use this to find your desired tank fragmentation value if you're still unsure. just @Redbert on discord if you have any questions.

Side issue, this does allow for assembly bombs to potentially become maxcaps with the right mix. I am confident I can hit the 8 devastation cap using your version of assembly bombs, I can get back to you later with a recipe if needed.

Also, I think one of the head maintainers were pretty set on having assembly bombs being extremely simple, so you might have trouble pushing that part through anyway.

**Comment by Pickle-Coding** [comment]
First three would probably be reasonable for ordnance. It's maxed pressure, so temperature inversely implies the mole requirement.

Also this assembly bomb simply heats the gas contents, so this is far simpler than the forest of if checks that it was before.

**Comment by GuillaumePrata** [comment]
With this change the assembly bombs are simpler than before.

The old system basically forced you to code dive, as it followed it's own logic, or get a recipe given to you

The new system follows ordnance knowledge, so it can can be transfered around from one to another, we have good tutorials and good reasons to learn how ordnance works too.
And I'm sure we will have people sharing recipes on the first day after merge so nothing changes on that.

**Comment by Deadgebert** [comment]
let me rephrase
maxcaps will be locked behind hypernob formation or 1:1:11 trit/hydrogen/oxy reaction. these are not very feasible through toxins for most players. 
assembly bombs were simple not because the code around it but because it's easy for people to approach. I agree that the knowledge of 'heat plasma to just above breakpoint and fill tank with 30 moles' is hard to find without code diving, but with that knowledge, the bomb is incredibly easy to make. You can do so without actual atmos knowledge. you can do it as an assistant with access to a plasma tank, a thermo machine, and an air pump.

past that, reactions in assembly bombs is pretty dangerous. with just a little bit of fiddling, i can already make a 5 dev assembly bomb on this PR with the following recipe.
58 oxygen : 5.15 hydrogen : 5.27 tritium
with more time, i'm sure other players can reach the 8 dev cap.

I guess my point is that this takes the knowledge requirement for proper bombs to be pushed far away from normal players, buffing me with my spreadsheets and 800 hours of atmos playtime. Science, research, and unlocks can easily be balanced around this change, but big booms are further from the average player.
